doc: update limitation about retain policy (#714)
topolvm/topolvm: docs/limitations.md

> The `csi` volume type does not support direct reference from Pod and may
> only be referenced in a Pod via a `PersistentVolumeClaim` object.

> If you delete a PVC with `Retain` reclaim policy, the corresponding `LogicalVolume` resource and the LVM logical volume are *NOT* deleted. If you delete this resource after that, the corresponding LVM logical volume is also deleted.

Reviewer: > If you delete a PVC with `Retain` reclaim policy,

PVCs do not have reclaim policy field, but StorageClasses and PVs do.

> If you delete this resource after that,

I feel it's not clear what "this resource" refers to.

My suggestion is as follows.

```suggestion
If you delete a PVC whose corresponding PV has `Retain` reclaim policy, the corresponding `LogicalVolume` resource and the LVM logical volume are *NOT* deleted. If you delete this `LogicalVolume` resource after deleting the PVC, the related LVM logical volume is also deleted.
```
